src/hssm/hssm.py

```diff
@@ -1618,26 +1700,156 @@ def _postprocess_initvals_deterministic(
+        if self.link_settings not in ["log_logit", None]:
+            _logger.info(
                 "Not preprocessing initial values, "
                 + "because none of the two standard link settings are chosen!"
             )
             return None
 
-        link_setting_str = str(self.link_settings)
         # Set initial values for particular parameters
         for name_, starting_value in self.pymc_model.initial_point().items():
-            name_tmp = name_
-            name_tmp = name_tmp.replace("_log__", "")
-            name_tmp = name_tmp.replace("_interval__", "")
-            if name_tmp in initval_settings[link_setting_str].keys():
+            # strip name of `_log__` and `_interval__` suffixes
+            name_tmp = name_.replace("_log__", "").replace("_interval__", "")
+
+            # We need to check if the parameter is actually backed by
+            # a regression.
+
+            # If not, we don't actually apply a link function to it as per default.
+            # Therefore we need to apply the initial value strategy corresponding
+            # to 'None' link function.
+
+            # If the user actively supplies a link function, the user
+            # should also have supplied an initial value insofar it matters.
+            if self.params[self._get_prefix(name_tmp)].is_regression:
+                param_link_setting = self.link_settings
+            else:
+                param_link_setting = None
+            if name_tmp in initval_settings[param_link_setting].keys():
+                if self._check_if_initval_user_supplied(name_tmp):
+                    _logger.info(
+                        "User supplied initial value detected for %s, \n"
+                        " skipping overwrite with default value.",
+                        name_tmp,
+                    )
+                    continue
+
                 # Apply specific settings from initval_settings dictionary
                 self.pymc_model.set_initval(
                     self.pymc_model.named_vars[name_tmp],
-                    initval_settings[link_setting_str][name_tmp],
+                    initval_settings[param_link_setting][name_tmp],
                 )
 
+    def _get_prefix(self, name_str: str) -> str:
+        """Get parameters wise link setting function from parameter prefix."""
+        # `p_outlier` is the only basic parameter floating around that has
+        # an underscore in it's name.
+        # We need to handle it separately. (Renaming might be better...)
+        if "_" in name_str:
+            if "p_outlier" not in name_str:
+                name_str_prefix = name_str.split("_")[0]
+            else:
+                name_str_prefix = "p_outlier"
+        else:
+            name_str_prefix = name_str
+        return name_str_prefix
+
+    def _check_if_initval_user_supplied(
+        self,
+        name_str: str,
+        return_value: bool = False,
+    ) -> Union[bool, float, int, np.ndarray, None]:
```

Comment:
For Python 3.10+ we use `|` instead of importing `Union`

Reply:
@digicosmos86 got rid of Union in multiple places now. 
The comment is without loss of generality?

Reply:
(I guess can't get rid of it everywhere got complaints when doing so in some places)

Reply:
Only place where it might complain is when the `|` operator is not used in type hints, for example in `isinstance` or `cast`, but as type hints you should be all set :)